### PR TITLE
add Dto level, add methods impl in repo and service, add mapper and c…

### PR DESCRIPTION
**Pull request by Alexander1iubchenko**
…onfig, add controller and custom exception

### CODE REVIEW

File: src/main/java/service/BookServiceImpl.java [hunk 1]
```diff
@@ -1,16 +1,35 @@
...
+import dto.CreateBookRequestDto;
+import exception.EntityNotFoundException;
 import java.util.List;
+import mapper.BookMapper;
 import model.Book;
+import repository.BookRepository;
 
 public class BookServiceImpl implements BookService {
+    private BookRepository bookRepository;
+    private BookMapper bookMapper;
```
Comment (kshuryhin) [inline src/main/java/service/BookServiceImpl.java]: I do not see where the initialisation of this fields is done. Also, good candidates to be final

File: src/main/java/com/example/bookstore/BookStoreApplication.java [hunk 2]
```diff
@@ -1,10 +1,7 @@
 package com.example.bookstore;
```
Comment (kshuryhin) [inline src/main/java/com/example/bookstore/BookStoreApplication.java]: This class should not be in this package I guess

**Review by kshuryhin — CHANGES_REQUESTED** [review]
Good job, but there are some points to consider. Also, consider better naming for git branches, which will be more feature-specific

File: src/main/java/controller/BookController.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    public List<BookDto> getAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.getBookById(id);
+    }
+
+    @GetMapping
```
Comment (d1sam) [inline src/main/java/controller/BookController.java]:
```suggestion
    @PostMapping
```

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but some critical changes should be done!

**Review by d1sam — APPROVED** [review]
well done!

**Pull request merged**